samples/javascript_nodejs/21.corebot-app-insights/index.js

```diff
@@ -33,18 +33,32 @@ const adapter = new BotFrameworkAdapter({
 });
 
 // Catch-all for errors.
-adapter.onTurnError = async (context, error) => {
-    // This check writes out errors to console log
+const onTurnErrorHandler = async (context, error) => {
+    // This check writes out errors to console log .vs. app insights.
     // NOTE: In production environment, you should consider logging this to Azure
     //       application insights.
-    console.error(`\n [onTurnError]: ${ error }`);
+    console.error(`\n [onTurnError] unhandled error: ${ error }`);
+
+    // Send a trace activity, which will be displayed in Bot Framework Emulator
+    await context.sendTraceActivity(
+        'OnTurnError Trace',
+        `${ error }`,
+        'https://www.botframework.com/schemas/error',
+        'TurnError'
+    );
+
     // Send a message to the user
-    const onTurnErrorMessage = 'Sorry, it looks like something went wrong!';
+    let onTurnErrorMessage = 'The bot encountered an error or bug.';
+    await context.sendActivity(onTurnErrorMessage, onTurnErrorMessage, InputHints.ExpectingInput);
+    onTurnErrorMessage = 'To continue to run this bot, please fix the bot source code.';
     await context.sendActivity(onTurnErrorMessage, onTurnErrorMessage, InputHints.ExpectingInput);
     // Clear out state
     await conversationState.delete(context);
 };
 
+// Set the onTurnError for the singleton BotFrameworkAdapter.
+adapter.onTurnError = onTurnErrorHandler;
+
 // Add telemetry middleware to the adapter middleware pipeline
 var telemetryClient = getTelemetryClient(process.env.InstrumentationKey);
 var telemetryLoggerMiddleware = new TelemetryLoggerMiddleware(telemetryClient, true);
```

Comment:
The second parameter for `TelemetryLoggerMiddleware` is `logPersonalInformation` - it is is a boolean value that is set to false by default.  The sample should not set this to `true`, and should probably leave it out all together since if left out, it defaults to false.  This is how it is done in the C# sample, and should be consistent in the JS sample.

```suggestion
var telemetryLoggerMiddleware = new TelemetryLoggerMiddleware(telemetryClient);
```

Reply:
Amended. Thanks.